### PR TITLE
RDoc-2111-Clarified intro statement about limitations and the limitat…

### PR DESCRIPTION
**Pull request by serezhub**
…ions section

### CODE REVIEW

File: Documentation/5.1/Raven.Documentation.Pages/client-api/bulk-insert/how-to-work-with-bulk-insert-operation.dotnet.markdown [hunk 1]
```diff
@@ -29,14 +44,28 @@ One of the features that is particularly useful when inserting large amount of d
...
-## Example
+* BulkInsert is designed to efficiently push high quantities of data.  
+  As such, data is streamed and **processed by the server in batches**.  
+  Each batch is fully transactional, but there are no transaction guarantees between the batches. The operation as a whole is non-transactional. 
+  If your bulk insert is interrupted mid-way, some of your data might be persisted on the server while some of it might not.  
+  * Make sure that your logic accounts for the possibility of an interruption where some of your data has not yet persisted on the server.
+  * If the operation was interrupted, one option is to re-insert the whole dataset in a new operation.  
+    It will overwrite existing documents.
+  * **If you need full transactionality**, the [session](../../client-api/session/what-is-a-session-and-how-does-it-work) may be a better option.  
+    If using the session, because all of the data is processed in one transaction, your machine's RAM must be able to handle the
```
Comment (ppekrol) [inline Documentation/5.1/Raven.Documentation.Pages/client-api/bulk-insert/how-to-work-with-bulk-insert-operation.dotnet.markdown:59]: your machine's RAM => your server resources

File: Documentation/5.1/Raven.Documentation.Pages/client-api/bulk-insert/how-to-work-with-bulk-insert-operation.dotnet.markdown [hunk 2]
```diff
@@ -29,14 +44,28 @@ One of the features that is particularly useful when inserting large amount of d
...
+  As such, data is streamed and **processed by the server in batches**.  
+  Each batch is fully transactional, but there are no transaction guarantees between the batches. The operation as a whole is non-transactional. 
+  If your bulk insert is interrupted mid-way, some of your data might be persisted on the server while some of it might not.  
+  * Make sure that your logic accounts for the possibility of an interruption where some of your data has not yet persisted on the server.
+  * If the operation was interrupted, one option is to re-insert the whole dataset in a new operation.  
+    It will overwrite existing documents.
+  * **If you need full transactionality**, the [session](../../client-api/session/what-is-a-session-and-how-does-it-work) may be a better option.  
+    If using the session, because all of the data is processed in one transaction, your machine's RAM must be able to handle the 
+    entire data-set included in the transaction.  
+* Bulk insert is **not thread-safe**, meaning that it isn't transactional.
```
Comment (ppekrol) [inline Documentation/5.1/Raven.Documentation.Pages/client-api/bulk-insert/how-to-work-with-bulk-insert-operation.dotnet.markdown:61]: thread-safe and transnationality are two different things. Remove ', meaning that it isn't transactional.  '

File: Documentation/5.2/Raven.Documentation.Pages/client-api/bulk-insert/how-to-work-with-bulk-insert-operation.dotnet.markdown [hunk 3]
```diff
@@ -1,8 +1,10 @@
 # Bulk Insert: How to Work With Bulk Insert Operation
```
Comment (ppekrol) [inline Documentation/5.2/Raven.Documentation.Pages/client-api/bulk-insert/how-to-work-with-bulk-insert-operation.dotnet.markdown:1]: Same comments in this file

File: Documentation/5.1/Raven.Documentation.Pages/client-api/bulk-insert/how-to-work-with-bulk-insert-operation.dotnet.markdown [hunk 4]
```diff
@@ -1,8 +1,21 @@
 # Bulk Insert: How to Work With Bulk Insert Operation
+---
 
-One of the features that is particularly useful when inserting large amount of data is `bulk inserting`. This is an optimized time-saving approach with few drawbacks that will be described later.
+{NOTE: }
 
-## Syntax
+* `BulkInsert` is useful when inserting a large quantity of data from the client to the server.  
+* It is an optimized time-saving approach with a few [limitations](../../client-api/bulk-insert/how-to-work-with-bulk-insert-operation#limitations)
+  such as transactionality and the minor possibility of interruptions during the operation.
```
[the remark this answers is not in the document]
Comment (serezhub) [reply Documentation/5.1/Raven.Documentation.Pages/client-api/bulk-insert/how-to-work-with-bulk-insert-operation.dotnet.markdown]: done

File: Documentation/5.1/Raven.Documentation.Pages/client-api/bulk-insert/how-to-work-with-bulk-insert-operation.dotnet.markdown (hunk 1 above)
Comment (serezhub) [reply Documentation/5.1/Raven.Documentation.Pages/client-api/bulk-insert/how-to-work-with-bulk-insert-operation.dotnet.markdown]: done

File: Documentation/5.1/Raven.Documentation.Pages/client-api/bulk-insert/how-to-work-with-bulk-insert-operation.dotnet.markdown (hunk 2 above)
Comment (serezhub) [reply Documentation/5.1/Raven.Documentation.Pages/client-api/bulk-insert/how-to-work-with-bulk-insert-operation.dotnet.markdown]: done

File: Documentation/5.2/Raven.Documentation.Pages/client-api/bulk-insert/how-to-work-with-bulk-insert-operation.dotnet.markdown (hunk 3 above)
Comment (serezhub) [reply Documentation/5.2/Raven.Documentation.Pages/client-api/bulk-insert/how-to-work-with-bulk-insert-operation.dotnet.markdown:1]: done

**Pull request merged**